workspace.spec

```diff
@@ -572,6 +572,378 @@ module Workspace {
+			when working on the resource.
+			Valid contributor_role values:
+			"Datacite:ContactPerson"
+			"Datacite:DataCollector"
+			"Datacite:DataCurator"
+			"Datacite:DataManager"
+			"Datacite:Distributor"
+			"Datacite:Editor"
+			"Datacite:HostingInstitution"
+			"Datacite:Producer"
+			"Datacite:ProjectLeader"
+			"Datacite:ProjectManager"
+			"Datacite:ProjectMember"
+			"Datacite:RegistrationAgency"
+			"Datacite:RegistrationAuthority"
+			"Datacite:RelatedPerson"
+			"Datacite:Researcher"
+			"Datacite:ResearchGroup"
+			"Datacite:RightsHolder"
+			"Datacite:Sponsor"
+			"Datacite:Supervisor"
+			"Datacite:WorkPackageLeader"
+			"Datacite:Other"
+			"CRediT:conceptualization"
+			"CRediT:data-curation"
+			"CRediT:formal-analysis"
+			"CRediT:funding-acquisition"
+			"CRediT:investigation"
+			"CRediT:methodology"
+			"CRediT:project-administration"
+			"CRediT:resources"
+			"CRediT:software"
+			"CRediT:supervision"
+			"CRediT:validation"
+			"CRediT:visualization"
+			"CRediT:writing-original-draft"
+			"CRediT:writing-review-editing"
+	*/
+	typedef structure {
+		string contributor_type;
+		string name;
+		string given_name;
+		string family_name;
+		string contributor_id;
+		list<Organization> affiliations;
+		list<string> contributor_roles;
+	} Contributor;
+
+	/*
+		Represents the credit metadata associated with a workspace object.
+
+		Required fields are:
+		- identifier
+		- resource_type
+		- titles (one or more required)
+		- contributors (one or more required)
+		- versioning information: if the resource does not have an explicit version number, one or more dates should be supplied, ideally the date of resource publication and the last update (if applicable).
+
+		string identifier - persistent unique identifier for the resource.
+			Should be in the format
+			<database name>:<identifier within database>
+			e.g.	DOI:10.46936/10.25585/60000745
+				CrossRef:IsManifestationOf
+				GO:0005456
+				HGNC:7470
+		string license (optional) - usage license for the resource.
+		string resource_type - the type of resource being represented.
+		string version (optional) - the version of the resource.
+		list<Contributor> contributors (optional) - a list of people and/or organizations who contributed to the resource.
+		list<EventDate> dates - a list of relevant lifecycle events for the resource.
+		list<FundingReference> funding (optional) - zero or more funding sources for the resource.
+		list<PermanentID> related_identifiers (optional) - zero or more PIDs related to the resource.
+		list<Title> titles - one or more titles for the resource.
+	*/
+	typedef structure {
+		string identifier;
+		string license;
+		string version;
+		string resource_type;
+		list<Contributor> contributors;
+		list<EventDate> dates;
+		list<FundingReference> funding;
+		list<PermanentID> related_identifiers;
+		list<Title> titles;
+	} CreditMetadata;
+
+	/* CreditMetadataContainer
```

Comment:
Given that at least the first version of this feature is going to be admin edits only, and the admin interface just takes `UnspecifiedObject`s, I don't think there's a need for this structure

Reply:
Removed.

Reply:
:+1: